includes/block-patterns/class-sensei-block-patterns.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Sensei Block Patterns.
+ *
+ * @package sensei-lms
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei Block Patterns class.
+ */
+class Sensei_Block_Patterns {
+	/**
+	 * Instance of class.
+	 *
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * Fetches the instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initializes the class.
+	 */
+	public function init() {
+		add_action( 'init', [ $this, 'register_block_patterns_category' ] );
+	}
+
+	/**
+	 * Sensei_Editor_Wizard constructor. Prevents other instances from being created outside of `self::instance()`.
+	 */
+	private function __construct() {
+	}
+
+	/**
+	 * Register Sensei block patterns category.
+	 *
+	 * @access private
+	 */
+	public function register_block_patterns_category() {
+		register_block_pattern_category(
+			self::get_patterns_category_name(),
+			[ 'label' => __( 'Sensei LMS', 'sensei-lms' ) ]
+		);
+	}
+
+	/**
+	 * Get patterns category name.
+	 */
+	public static function get_patterns_category_name() {
+		return 'sensei-lms';
```

Comment:
I moved from the constants to be easier to deprecate and so on. I thought to use a private constant, but it's not supported in PHP 7.0.